source/extensions/filters/http/transformation/transformation_filter.cc

```diff
@@ -139,47 +139,25 @@ TransformationFilter::encodeTrailers(Http::HeaderMap &) {
   return Http::FilterTrailersStatus::Continue;
 }
 
-void TransformationFilter::checkRequestActive() {
+void TransformationFilter::setupTransformationPair() {
   route_ = decoder_callbacks_->route();
-  request_transformation_ =
-      getTransformFromRoute(TransformationFilter::Direction::Request);
-}
-
-void TransformationFilter::checkResponseActive() {
-  response_transformation_ =
-      getTransformFromRoute(TransformationFilter::Direction::Response);
-}
-
-TransformerConstSharedPtr TransformationFilter::getTransformFromRoute(
-    TransformationFilter::Direction direction) {
-
-  if (!route_) {
-    return nullptr;
-  }
 
   const auto *route_config = Http::Utility::resolveMostSpecificPerFilterConfig<
-      RouteTransformationFilterConfig>(filter_config_->name(), route_);
-  
-  switch (direction) {
-    case TransformationFilter::Direction::Request: {
-      should_clear_cache_ = filter_config_->shouldClearCache();
-      if (route_config != nullptr && route_config->getRequestTranformation() != nullptr) {
-        should_clear_cache_ = route_config->shouldClearCache();
-        return route_config->getRequestTranformation();
-      } else {
-        return filter_config_->getRequestTranformation();
-      }
-    }
-    case TransformationFilter::Direction::Response: {
-      if (route_config != nullptr && route_config->getResponseTranformation() != nullptr) {
-        return route_config->getResponseTranformation();
-      } else {
-        return filter_config_->getResponseTranformation();
-      }
-    }
+    RouteTransformationFilterConfig>(filter_config_->name(), route_);
+
+  // if there is a route level config present, automatically disregard header_matching rules
+  if (route_config != nullptr) {
+    has_route_level_config_ = true;
+    should_clear_cache_ = route_config->shouldClearCache();
```

Comment:
shouldClearCache should probably be in the else flow? probably part of the TransformerPairConstSharedPtr?